### PR TITLE
[Event Hubs Client] Track Two: Second Preview (Samples)  …

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The intent of these changes is enhance the samples to cover the new scenarios added for preview two and improve upon the existing set of samples.

# Last Upstream Rebase

Tuesday, August 6, 2019  9:090am (EDT)

# Resources

- [Azure Messaging Exception Documentation](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions)
- [.NET Event Hubs Client: Second Preview Design Proposal](https://github.com/jsquire/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md)

# Related and Follow-Up Issues

- [[Epic] Release Event Hubs Track 2 Library Preview 2](https://github.com/Azure/azure-sdk-for-net/issues/6752) (#6752)
- [Create samples](https://github.com/Azure/azure-sdk-for-net/issues/6797) (#6797)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ConsumeEventsWithMaximumWaitTime.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                    // available for our consumer to see.  We will iterate over the available events in the partition, which should be
+                    // just the events that we published.
+                    //
+                    // When a maximum wait time is specified, the iteration will ensure that it returns control after that time has elapsed,
+                    // whether or not an event is available in the partition.  If no event was available a null value will be emitted instead.
+                    // This is intended to return control to the loop and avoid blocking for an indeterminate period of time to allow event
+                    // processors to verify that the iterator is still consuming the partition and to make decisions on whether or not to continue
+                    // if events are not arriving.
+                    //
+                    // For this example, we will specify a maximum wait time, and won't exist the loop until we've received at least one more
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ConsumeEventsWithMaximumWaitTime.cs]: typo: and won't **exit** the loop

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 2]
```diff
@@ -210,32 +210,43 @@ Each of the samples is self-contained and focused on illustrating one specific s
...
+- [Hello world](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample01_HelloWorld.cs)  
   An introduction to Event Hubs, illustrating how to connect and query the service.
 
-- [Create an Event Hub client with custom options](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample2_ClientWithCustomOptions.cs)  
+- [Create an Event Hub client with custom options](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample02_ClientWithCustomOptions.cs)  
   An introduction to Event Hubs, exploring additional options for creating an Event Hub client.
   
-- [Publish an event to an Event Hub](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample3_PublishAnEvent3.cs)  
+- [Publish an event to an Event Hub](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample3_PublishAnEvent03.cs)
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Shouldn't it be `Sample03_PublishAnEvent`?

**Review by kinelski — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Good catch.  That got mangled!

**Pull request merged**